pkg/collector/container.go

```diff
@@ -33,6 +35,22 @@ func Container(cfg config.Config, logger logr.Logger, otelcol v1alpha1.OpenTelem
 		image = cfg.CollectorImage()
 	}
 
+	// build port map
+	ports := getConfigContainerPorts(logger, otelcol.Spec.Config)
+	for _, p := range otelcol.Spec.ContainerPorts {
+		if len(p.Name) > 15 {
+			p.Name = p.Name[:15]
+		}
+		if len(validation.IsValidPortName(p.Name)) == 0 && len(validation.IsValidPortNum(int(p.ContainerPort))) == 0 {
+			ports[p.Name] = p
+		}
+	}
+	ports["metrics"] = corev1.ContainerPort{
+		Name:          "metrics",
+		ContainerPort: 8888,
```

Comment:
iirc some people override the metrics port, we may not want to hardcode what the value is here

Reply:
We can pull the port from `spec.config.service.telemetry.metrics.address` i believe